x-pack/plugins/alerting/server/application/r_rule/schemas/r_rule_schema.ts

```diff
@@ -34,13 +34,15 @@ export const rRuleSchema = schema.object({
       schema.literal('SU'),
     ])
   ),
-  byweekday: schema.maybe(schema.arrayOf(schema.oneOf([schema.string(), schema.number()]))),
-  bymonth: schema.maybe(schema.arrayOf(schema.number())),
-  bysetpos: schema.maybe(schema.arrayOf(schema.number())),
-  bymonthday: schema.maybe(schema.arrayOf(schema.number())),
-  byyearday: schema.maybe(schema.arrayOf(schema.number())),
-  byweekno: schema.maybe(schema.arrayOf(schema.number())),
-  byhour: schema.maybe(schema.arrayOf(schema.number())),
-  byminute: schema.maybe(schema.arrayOf(schema.number())),
-  bysecond: schema.maybe(schema.arrayOf(schema.number())),
+  byweekday: schema.maybe(
+    schema.nullable(schema.arrayOf(schema.oneOf([schema.string(), schema.number()])))
+  ),
+  bymonth: schema.maybe(schema.nullable(schema.arrayOf(schema.number()))),
```

Comment:
I am also not sure about this part, and I hope @Zacqary can help us here.
Because these fields are nullable in the `rrule` lib but not nullable and optional in everywhere else. We should align them. Have we ever saved them as null?

Reply:
I don't recall us ever explicitly saving these values as `null` but it shouldn't create any issues to change this.